js/interface.js

```diff
@@ -877,24 +898,23 @@ $('#app')
     $('[data-show-all-source]').addClass('hidden');
     $('[data-app-source]').removeClass('hidden');
 
-    if ($('[data-show-trash-source]').hasClass('active-source')) {
-      isShowingAll = false;
-      $('[data-show-trash-source]').click();
-    } else {
-      isShowingAll = true;
+    isShowingAll = true;
 
+    if (isShowingTrash) {
+      getTrashedDataSources();
+    } else {
       getDataSources();
     }
   })
   .on('click', '[data-app-source]', function() {
     $('[data-app-source]').addClass('hidden');
     $('[data-show-all-source]').removeClass('hidden');
 
-    if ($('[data-show-trash-source]').hasClass('active-source')) {
-      $('[data-show-trash-source]').click();
-    } else {
-      isShowingAll = false;
+    isShowingAll = false;
 
+    if (isShowingTrash) {
```

Comment:
couldn't you simply have `getDataSources()` to take care of whether it needs to fetch trashed or non trashed data sources? You literally have the `isShowingTrash` variable for it.

Reply:
@squallstar Yes, we could combine getTrashedDataSources and getDataSources but the API that we are using to get trashed and alive data sources return different results and in that case, we will need to use a separate promise chain and it will make the function much larger and harder to read than now.
Should we do that?